### DISCUSSION TITLE
0.7 migration question, what happened to with! macro?

### DISCUSSION THREAD

**Discussion by nlfiedler** [leptos-rs/leptos#3335] [General]
It is completely gone, no mention in the release notes. What do I do instead?

**Comment by gbj** [comment]
It was probably added some time during the process that I was porting things, and I forgot to add it back.

Here are some options:
1) Use `.read()`, which is new and serves much of the same purpose of avoiding the indentation creep of nested `.with()`
2) Find the old `with!()` macro on the `leptos_0.6` branch and use it in your old project 
3) Find the old `with!()` macro on the `leptos_0.6` branch and make a PR to add it back to 0.7

**Comment by nlfiedler** [comment]
I'd love to contribute, if I wasn't struggling just using leptos. :)

Thank you.
